pilot/pkg/serviceregistry/kube/controller/ambient/ambientindex.go

```diff
@@ -138,7 +138,10 @@ func New(options Options) Index {
 	GatewayClasses := krt.WrapClient[*v1beta1.GatewayClass](gatewayClassClient, krt.WithName("GatewayClasses"))
 
 	Services := krt.NewInformerFiltered[*v1.Service](options.Client, filter, krt.WithName("Services"))
-	Nodes := krt.NewInformerFiltered[*v1.Node](options.Client, filter, krt.WithName("Nodes"))
+	Nodes := krt.NewInformerFiltered[*v1.Node](options.Client, kclient.Filter{
+		ObjectFilter:    options.Client.ObjectFilter(),
+		ObjectTransform: kubeclient.StripNodeUnusedFields,
```

Comment:
it's shared if we have the same inputs. We don't currently, and the test detects that and fails. but the test only runs in postsubmit. 

So this PR ensures we share

Reply:
Since the underlying informer is shared, if at begining a informer is initialized with a very strict `ObjectTransform`, later we want a informer with a loose transformer, may not want to strip any field. In this case the informer factory currently works unexpected.

Reply:
So this line and the `c.nodes =...` both pcreate a `kclient` which creates an informer from the informer factory. 

The informer factory will:
* If all filter is the same, share
* If filter is different, create a new one. In assertion test, `panic` when this happens

It is true we may end up with a case where 2 parts of the code have different requirements. In order to share informers, we need to take into account the global context of the program and ensure all informers have consistent filters. For example, if ambient needed all the fields, we would need to go to _both_ places and remove the `ObjectTransform`.

This is not ideal from a code maintenance perspective, but ensures perfect informer sharing.